src/components/filters/DateFilter/index.tsx

```diff
@@ -35,34 +37,81 @@ const FieldWrapper = styled.div`
 const Date = styled(Textfield)`
   font-size: 14px;
   height: 32px;
+  padding-left: 6px;
+  padding-right: 6px;
 `
 
 interface Props {
-  onChangeFrom: () => void
-  onChangeTo: () => void
-  onSubmit: () => void
+  onChangeFrom?: (event: React.ChangeEvent<HTMLInputElement>) => void
+  onChangeTo?: (event: React.ChangeEvent<HTMLInputElement>) => void
+  onSubmit: (from: number, to: number) => void
   title: string
 }
 
 export const DateFilter: React.FC<Props> = (props) => {
   const { onChangeFrom, onChangeTo, onSubmit, title } = props
 
+  const [from, setFrom] = React.useState<Maybe<number>>(null)
+  const [to, setTo] = React.useState<Maybe<number>>(null)
+
+  const onChangeFromInternal = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (typeof onChangeFrom === 'function') {
+      onChangeFrom(event)
+    }
+
+    const fromDateMoment = moment(event.currentTarget.value)
+    const fromTimestamp = fromDateMoment.unix()
+
+    setFrom(fromTimestamp)
+  }
+
+  const onChangeToInternal = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (typeof onChangeTo === 'function') {
+      onChangeTo(event)
+    }
+
+    const toDateMoment = moment(event.currentTarget.value)
+    const toTimestamp = toDateMoment.unix()
```

Comment:
This could be changed to make all the conversion without using extra variables, `moment(...).unix()` but also moved to a helper wich make explicit in his name what you are doing, `dateToTheGraphFormat`.

Reply:
looking in the code, this is the only place where something similar is used. I think at the moment it doesn't make sense